app/parse.py

```diff
@@ -15,9 +39,78 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+def parse_category(category_url: str) -> list[Product]:
+    driver.get(category_url)
+    products = []
+    try:
+        while True:
+            try:
+                element = WebDriverWait(driver, 2).until(
+                    ec.presence_of_element_located((By.CLASS_NAME, MORE_CLASS))
+                )
+                if element.is_displayed() and element.is_enabled():
+                    element.click()
+                else:
+                    break
+            except (NoSuchElementException, ElementNotInteractableException):
+                break
+    except:
+        pass
```

Comment:
The exception handling here is too broad. Catching all exceptions with a bare `except` can mask errors that should be addressed. Consider catching specific exceptions or logging the error for better debugging.